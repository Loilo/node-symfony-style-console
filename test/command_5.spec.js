

const wrapper = require('./wrapper')

test('05 Ensure has proper line ending before outputting a text block like with SymfonyStyle.listing() or SymfonyStyle.text()', wrapper(io => {
    io.writeln('Lorem ipsum dolor sit amet')
    io.listing([
        'Lorem ipsum dolor sit amet',
        'consectetur adipiscing elit',
    ])

    // Even using write:
    io.write('Lorem ipsum dolor sit amet')
    io.listing([
        'Lorem ipsum dolor sit amet',
        'consectetur adipiscing elit',
    ])

    io.write('Lorem ipsum dolor sit amet')
    io.text([
        'Lorem ipsum dolor sit amet',
        'consectetur adipiscing elit',
    ])

    io.newLine()

    io.write('Lorem ipsum dolor sit amet')
    io.comment([
        'Lorem ipsum dolor sit amet',
        'consectetur adipiscing elit',
    ])
}))
