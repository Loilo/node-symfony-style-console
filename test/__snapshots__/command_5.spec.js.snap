// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`05 Ensure has proper line ending before outputting a text block like with SymfonyStyle.listing() or SymfonyStyle.text() 1`] = `
"Lorem ipsum dolor sit amet
 * Lorem ipsum dolor sit amet
 * consectetur adipiscing elit

Lorem ipsum dolor sit amet
 * Lorem ipsum dolor sit amet
 * consectetur adipiscing elit

Lorem ipsum dolor sit amet
 Lorem ipsum dolor sit amet
 consectetur adipiscing elit

Lorem ipsum dolor sit amet

 // Lorem ipsum dolor sit amet                                                                                          
 //                                                                                                                     
 // consectetur adipiscing elit                                                                                         

"
`;
